### PR TITLE
[enhancement](compaction) add delete_sign_index check before filter delete

### PR DESCRIPTION
**Pull request by yixiutt**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/olap/block_reader.cpp [hunk 1]
```diff
@@ -280,6 +280,11 @@ Status BlockReader::_unique_key_next_block(Block* block, MemPool* mem_pool, Obje
     if (_filter_delete) {
         int delete_sign_idx = _reader_context.tablet_schema->field_index(DELETE_SIGN);
         DCHECK(delete_sign_idx > 0);
+        if (delete_sign_idx <= 0 || delete_sign_idx >= target_columns.size()) {
```
Comment (liaoxin01) [inline be/src/vec/olap/block_reader.cpp:283]: delete line 282 ?

Comment (yixiutt) [reply be/src/vec/olap/block_reader.cpp:283]: maybe we can find some bug with asan use DCEHCK

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.97 seconds
 load time: 427 seconds
 storage size: 17123356954 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221221031236_clickbench_pr_65895.html

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**